string log_dir_path
---